sockshop/payment/cmd/paymentsvc/main.go

```diff
@@ -23,6 +23,25 @@ func main() {
 	errc := make(chan error)
 	ctx := context.Background()
 
+	handler, logger := Handler(ctx, float32(*declineAmount))
+
+	// Create and launch the HTTP server.
+	go func() {
+		logger.Log("transport", "HTTP", "port", *port)
+		errc <- http.ListenAndServe(":"+*port, handler)
+	}()
+
+	// Capture interrupts.
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+		errc <- fmt.Errorf("%s", <-c)
+	}()
+
+	logger.Log("exit", <-errc)
+}
+
+func Handler(ctx context.Context, declineAmount float32) (http.Handler, log.Logger) {
```

Comment:
Hmm, the name is a little confusing here, this isn't just returning an http.Handler, it's returning all of the components in the object graph. Maybe Build or WireUp or something like this makes more sense?